src/LineageTree/lineageTree.py

```diff
@@ -2400,6 +2403,595 @@ def get_cells_at_t_from_root(self, r: [int, list], t: int = None) -> list:
             return list(r)
         return final_nodes
 
+    @staticmethod
+    def __dp(dist_mat, start_d=0, back_d=0):
+        """
+        Find DTW minimum cost between two series using dynamic programming.
+
+            Args:
+                dist_mat (matrix): distance matrix obtained by the function calculate_dtw
+                start_d (int): start delay
+                back_d (int): end delay
+
+            Returns:
+                (tuple of tuples) Aligment path
+                (matrix) Cost matrix
+        """
+        N, M = dist_mat.shape
+
+        # Initialize the cost matrix
+        cost_mat = np.zeros((N + 1, M + 1))
+        cost_mat[0, 1:] = np.inf
+        cost_mat[1:, 0] = np.inf
+        cost_mat[0, 0] = 0
+
+        # Fill the cost matrix while keeping traceback information
+        traceback_mat = np.zeros((N, M))
+
+        cost_mat[: start_d + 1, 0] = 0
+        cost_mat[0, : start_d + 1] = 0
+
+        cost_mat[N - back_d :, M] = 0
+        cost_mat[N, M - back_d :] = 0
+
+        for i in range(N):
+            for j in range(M):
+                penalty = [
+                    cost_mat[i, j],  # match (0)
+                    cost_mat[i, j + 1],  # insertion (1)
+                    cost_mat[i + 1, j],
+                ]  # deletion (2)
+                i_penalty = np.argmin(penalty)
+                cost_mat[i + 1, j + 1] = dist_mat[i, j] + penalty[i_penalty]
+                traceback_mat[i, j] = i_penalty
+
+        min_index1 = np.argmin(cost_mat[N - back_d :, M])
+        min_index2 = np.argmin(cost_mat[N, M - back_d :])
+
+        if (
+            cost_mat[N - back_d + min_index1, M]
```

Comment:
I rather always use the symbole `<` so the smaller element is always to the left.